### ISSUE TITLE
Open In Code does not open file/folder in devcontainer if already open

### ISSUE THREAD

**Issue by ADustyOldMuffin** [microsoft/vscode#127587]
Does this issue occur when all extensions are disabled?: Yes

![image](https://user-images.githubusercontent.com/41391321/123884719-83923280-d911-11eb-82b8-664b004b23d9.png)

Steps to Reproduce:

1. Open up repository in devcontainer
2. browse in file explorer to folder
3. Right click -> Open in Code, File will open in new VS Code window

This is referenced in an issue [here](https://github.com/desktop/desktop/issues/11634) and I just wanted to check and see if this is expected or if it's an actual bug.

**Comment by isidorn** [comment]
If I understand you correctly: we do not allow to open the same folder in two windows. There is a long standing vscode issue which you can find. And I am closing this as a duplicate of that one

fyi @chrmarti for dev containers

**Comment by ADustyOldMuffin** [comment]
> If I understand you correctly: we do not allow to open the same folder in two windows.

@isidorn Yes and this is what I'm saying happens, I can have a dev container open but then I try and click open in code with a file or even the same folder as the dev container it will open up a new window outside the dev container.

**Comment by ADustyOldMuffin** [comment]
See the behavior below, I know that since it's running in a container the file system is "different" so I don't know if that makes a difference. I just wanted to confirm if this works as expected or if this is a bug.

![vscode-behavior](https://user-images.githubusercontent.com/41391321/123974535-47e38100-d982-11eb-8c33-4db9821ad35e.gif)

**Comment by chrmarti** [comment]
That's expect, the same files/folders have URIs inside the container that are different from the URIs on the local filesystem. For VS Code it looks like different files/folders.

**Comment by ADustyOldMuffin** [comment]
Gotcha, and that's what I figured I just wanted to confirm it for that other issue.

**Comment by Soneji** [comment]
@chrmarti would you be able to link us to the other issue. From what you explain of the other issue, I don't think this is a duplicate.

I think this could be remedied by detecting the URLs and matching them up and opening it in the correct window? 

It is quite annoying when you're working on a project in a devcontainer and this happens. This is also the case if I have only one window of vscode with /folder open in a devcontainer and I open /folder/file1, it will open a brand new window and open that file1 in there. Surely the best behaviour would be to open file1 within the devcontainer?

Thanks

**Comment by chrmarti** [comment]
There is currently no way for VS Code to find out if the same file can be opened in an existing window just with a different URI.

There is now a way for remote resolvers to tell VS Code what the canonical URI of a given URI is, but that's the reverse mapping.

@bpasero for your consideration.

**Comment by Soneji** [comment]
What do you mean by “remote resolvers”?

**Comment by chrmarti** [comment]
E.g., Remote-Containers and Remote-WSL. Both can open some local files as "remote" files because they are mounted into the remote filesystem.

**Comment by bpasero** [comment]
I am not sure I fully understand: you are in a remote window and want to open local files? I think it is "by design" that a window with "remote authority" cannot open any local files, assigning to @aeschli who introduced this concept.

**Comment by Soneji** [comment]
I think you misunderstand.

Let me write a flow for you to replicate/understand the issue:
Setup:
- You open a project that has a `.devcontainer` folder, and vscode asks you to "reopen in devcontainer", so you open the devcontainer. 
- You now have one vscode window of a project in a devcontainer. 

Reproducing:
- You navigate to another application, such as file explorer or GitHub desktop and find a file in the project you're working on, which exists in the devcontainer. 
- You right click on the file, and you can choose an option in windows explorer to "Open with code" or in GitHub desktop "open in Visual Studio Code". 
- It doesn't matter which option you take, you click the menu item. 
- (I reckon it would also be an issue for just generally opening filetypes that have code as the default opener)

Behaviour:
- vscode will now open this file in a brand new vscode window, outside of the devcontainer. 
- We think the preferred behaviour should be for vscode to detect that this file the user has requested to be opened exists in both the filesystem and in the open devcontainer, and translate that file path to devcontainer/WSL language and open it in the existing devcontainer/remote wsl window.

**Comment by bpasero** [comment]
Thanks, this is very unlikely to happen because I doubt we have this kind of knowledge when opening a file, but anyway leaving open as feature request.

**Comment by Soneji** [comment]
@ADustyOldMuffin would you be able to update the original post with my description [here](https://github.com/microsoft/vscode/issues/127587#issuecomment-896880482)
Let's hope we can get 20 upvotes
